org.eclipse.xtext.util/src/org/eclipse/xtext/util/ITextRegion.java

```diff
@@ -17,11 +17,43 @@ public interface ITextRegion {
 
 	int getLength();
 
+	/**
+	 * @since 2.41
+	 */
+	default int getEndOffset() {
+		return getOffset() + getLength();
+	};
+
 	ITextRegion merge(ITextRegion region);
 	
 	boolean contains(ITextRegion other);
 	
 	boolean contains(int offset);
 
-	static ITextRegion EMPTY_REGION = ITextRegionWithLineInformation.EMPTY_REGION;
+	static ITextRegion EMPTY_REGION = new ITextRegion() {
```

Comment:
@cdietrich , @szarnekow , this is a bit tricky. I need to remove the dependency to ITextRegionWithLineInformation because such a circular dependency does not work with the default method. There is an exception and `EMPTY_REGION` is set to `null`. 
I think the modification is fine, but if you are worried about possible consumers using `==` and not `equals` I would not know how to implement `getEndOffset`, and I would fallback to not do it.